internal/grpc/services/gateway/storageprovider.go

```diff
@@ -308,166 +305,30 @@ func (s *svc) getHome(_ context.Context) string {
+	_, st := s.getPath(ctx, req.Ref)
 	if st.Code != rpc.Code_CODE_OK {
 		return &gateway.InitiateFileDownloadResponse{
 			Status: st,
 		}, nil
 	}
 
-	if !s.inSharedFolder(ctx, p) {
-		statReq := &provider.StatRequest{Ref: req.Ref}
-		statRes, err := s.stat(ctx, statReq)
-		if err != nil {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+statReq.Ref.String()),
-			}, nil
-		}
-		if statRes.Status.Code != rpc.Code_CODE_OK {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: statRes.Status,
-			}, nil
-		}
-		return s.initiateFileDownload(ctx, req)
-	}
-
-	if s.isSharedFolder(ctx, p) {
-		log.Debug().Str("path", p).Msg("path points to shared folder")
-		err := errtypes.PermissionDenied("gateway: cannot download share folder: path=" + p)
-		log.Err(err).Msg("gateway: error downloading")
+	statReq := &provider.StatRequest{Ref: req.Ref}
+	statRes, err := s.stat(ctx, statReq)
+	if err != nil {
 		return &gateway.InitiateFileDownloadResponse{
-			Status: status.NewInvalidArg(ctx, "path points to share folder"),
+			Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+statReq.Ref.String()),
 		}, nil
-
 	}
-
-	if s.isShareName(ctx, p) {
-		statReq := &provider.StatRequest{Ref: req.Ref}
-		statRes, err := s.stat(ctx, statReq)
-		if err != nil {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error stating ref:"+statReq.Ref.String()),
-			}, nil
-		}
-		if statRes.Status.Code != rpc.Code_CODE_OK {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: statRes.Status,
-			}, nil
-		}
-
-		if statRes.Info.Type != provider.ResourceType_RESOURCE_TYPE_REFERENCE {
-			err := errtypes.BadRequest(fmt.Sprintf("gateway: expected reference: got:%+v", statRes.Info))
-			log.Err(err).Msg("gateway: error stating share name")
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewInternal(ctx, err, "gateway: error initiating download"),
-			}, nil
-		}
-
-		ri, protocol, err := s.checkRef(ctx, statRes.Info)
-		if err != nil {
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewStatusFromErrType(ctx, "error resolving reference "+statRes.Info.Target, err),
-			}, nil
-		}
-
-		if protocol == "webdav" {
-			// TODO(ishank011): pass this through the datagateway service
-			// For now, we just expose the file server to the user
-			ep, opaque, err := s.webdavRefTransferEndpoint(ctx, statRes.Info.Target)
-			if err != nil {
-				return &gateway.InitiateFileDownloadResponse{
-					Status: status.NewInternal(ctx, err, "gateway: error downloading from webdav host: "+p),
-				}, nil
-			}
-			return &gateway.InitiateFileDownloadResponse{
-				Status: status.NewOK(ctx),
-				Protocols: []*gateway.FileDownloadProtocol{
-					{
-						Opaque:           opaque,
-						Protocol:         "simple",
-						DownloadEndpoint: ep,
-					},
-				},
-			}, nil
-		}
```

Comment:
@ishank011 there are several places in the gateway that have special "webdav"  handling ... what for? AFAICT it is related to OCM. Unfortunately, there are no tests covering that right now. Well the oc10 core testsuite does cover federated sharing, but the drone ci does not set up a second instance ... 

While there are ways forward to cover OCM let's start with what these webdav handling is meant to do? Could you enlighten us?

Reply:
@butonic yep, the OCM share targets are saved as `webdav://$token@$idp?name=$path`. When OCM shares are accepted, the references have these targets instead of `cs3` for other shares.
When `checkRef` is called and it detects that the reference is of type 'webdav', the requests are then forwarded to the methods in the `webdavstorageprovider.go` file, and they actually make the webdav calls to the idp of the share creator.

Reply:
@ishank ... um `$path`? so the references will break when the path changes? Why are we not using ids? Is OCM necessary for your current deployment?

Reply:
IMO the share storage provider will return a cs3 reference which resolves to a yet to be implemented OCM storage provider. That can encapsulate the token and idp data necessary to access the remote instance.

Reply:
I am currently looking at the `/examples/ocmd` setup. AFAICT it is deployed next to an existing reva deployment ... is it only starts the ocmd service on the http port ... so normal webdav is not available without an additional reva deployment.

Reply:
I started a PR to clean up the toml configs and add an ocm config set in https://github.com/cs3org/reva/pull/2239